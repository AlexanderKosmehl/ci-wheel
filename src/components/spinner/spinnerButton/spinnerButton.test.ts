import {
  describe, expect, it, jest,
} from '@jest/globals';
import generateSpinnerButton from './spinnerButton';

describe('generateSpinnerButton', () => {
  const onClick = jest.fn(() => {});
  const buttonComponent = generateSpinnerButton({ onClick });

  const button = buttonComponent.querySelector<HTMLButtonElement>('div > button');

  it('generates component correctly', () => {
    expect(buttonComponent).toMatchSnapshot();
  });

  it('adds working clickListener', () => {
    if (!button) throw Error('No button rendered!');

    button.click();

    expect(onClick).toBeCalled();
  });
});
