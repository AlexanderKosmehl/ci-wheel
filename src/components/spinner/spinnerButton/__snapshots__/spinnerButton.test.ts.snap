// Vitest Snapshot v1

exports[`generateSpinnerButton > generates component correctly 1`] = `
<button
  class="button"
  data-test="spinnerButton"
>
  <img
    class="icon"
    src="/src/assets/rotate-icon.svg"
  />
</button>
`;
